src/tools/rbd_wnbd/rbd_wnbd.cc

```diff
@@ -695,14 +700,17 @@ int disconnect_all_mappings(
           derr << "Could not remove mapping: " << cfg.devpath
                << ". Error: " << r << dendl;
         } else {
-          dout(5) << "Successfully removed mapping: " << cfg.devpath << dendl;
+          dout(1) << "Successfully removed mapping: " << cfg.devpath << dendl;
         }
       });
   }
   pool.join();
 
   r = iterator.get_error();
-  if (r) {
+  if (r == ENOENT) {
+    dout(0) << __func__ << ": wnbd adapter unavailable, "
+            << "assuming that no wnbd mappings exist." << dendl;
```

Comment:
> About the data race, we're simply assigning any non-zero error code, which should be ok without serializing the access to that value. Please let me know if I'm missing something.

It's probably OK but a code smell.  And it could be flagged by a data race sanitizer (maybe -- we aren't actually running any).

Reply:
I've updated the lambda capturing, explicitly specifying which variables to capture. ``err`` is now the only captured reference and is wrapped using ``std::atomic``.